Lib/bdb.py

```diff
@@ -84,6 +85,9 @@ def trace_dispatch(self, frame, event, arg):
 
         The arg parameter depends on the previous event.
         """
+
+        self.enterframe = frame
```

Comment:
does this change the lifetime of `frame`?

Reply:
Ah, interesting observation. Yes this holds a reference to `frame` (but only one) and could potentially extend the lifetime of it. The existing code in `main` has this issue as well. Not sure how serious this is - it's not ideal, but it will only hold reference to one frame. We could do a weakref here (but we need to do `self.enterframe()` because it's a ref now and that's not great), or we could reset it somewhere (but we need to change plenty of the structure of the dispatching code), or we could just not patch this in 3.12.

Reply:
You hold a ref to one frame, but this frame links to all the others throughout f_back. It can be a lot. 

What's the scenario where this is the only reference to this frame?

Reply:
When the program runs but `bdb` is left there - because we never clear this. There are other frames that we keep holding in `bdb` like `self.botframe` and `self.stopframe`. One solution could be clear all of them in `set_continue` when there's no breakpoint - so if the user continues the program without debugger overhead, all the frame references will be released. I think that might worth a separate PR.